server/mpm/event/event.c

```diff
@@ -1921,37 +1961,40 @@ static void * APR_THREAD_FUNC listener_thread(apr_thread_t * thd, void *dummy)
         }
 
         /* Same for queues, use their next expiry, if any. */
-        timeout_time = queues_next_expiry;
-        if (timeout_time
-                && (timeout_interval < 0
-                    || timeout_time <= now
-                    || timeout_interval > timeout_time - now)) {
-            timeout_interval = timeout_time > now ? timeout_time - now : 1;
+        expiry = queues_next_expiry;
+        if (expiry
+                && (timeout < 0
+                    || expiry <= now
+                    || timeout > expiry - now)) {
+            timeout = expiry > now ? expiry - now : 0;
```

Comment:
Why now `0` in case `expiry <= now` instead of `1` before?

Reply:
Below rounding up to milliseconds checks for timeout > 0, which shouldn't apply here.
Here we can afford a nonblocking poll() if the timeout already expired, this is `0` (`1` was almost that, I put `1` before here because I wasn't sure that `0` would work as expected, but it seems to by reading apr_pollset implementations).

Reply:
ok